src/create.rs

```diff
@@ -520,18 +584,65 @@ impl Execution {
 
         // Compile the installer
         info!("Compiling the installer");
-        let mut compiler = self.compiler()?;
+
+        let mut compiler = if !self.wix_toolset.is_modern() {
+            self.compiler()?
+        } else {
+            debug!("Using modern wix build tools");
+            let mut wix = Command::new("wix");
+            wix.arg("build");
+            wix
+        };
         debug!("compiler = {:?}", compiler);
+
+        if self.wix_toolset.is_modern() {
+            match &self.wix_toolset_upgrade {
+                WixToolsetUpgrade::None => {
+                    debug!("No toolset upgrade mode is set");
+                },
+                WixToolsetUpgrade::Inplace | WixToolsetUpgrade::InplaceVendored | WixToolsetUpgrade::SideBySide => {
+                    debug!("Starting toolset upgrade checks");
+                    let mut upgrade = crate::upgrade::WixUpgrade::try_start_upgrade()?;
+
+                    for wxs in wxs_sources.iter() {
+                        upgrade.add_wxs_source(wxs.to_path_buf())?;
+                    }
+
+                    debug!("Starting source file conversions");
+                    upgrade.convert(if matches!(self.wix_toolset_upgrade, WixToolsetUpgrade::SideBySide) {
+                        let current = std::env::current_dir()?;
+                        let sxs_folder = current.join(upgrade.sxs_folder_name());
+                        std::fs::create_dir_all(&sxs_folder)?;
+                        Some(sxs_folder)
+                    } else { 
+                        None
+                    })?;
+
+                    debug!("Installing any missing extensions");
+                    upgrade.install_extensions(matches!(self.wix_toolset_upgrade, WixToolsetUpgrade::Inplace), if matches!(self.wix_toolset_upgrade, WixToolsetUpgrade::SideBySide) {
+                        let current = std::env::current_dir()?;
+                        let sxs_folder = current.join(upgrade.sxs_folder_name());
+                        std::fs::create_dir_all(&sxs_folder)?;
+                        Some(sxs_folder)
+                    } else { 
+                        None
+                    })?;
+                },
+            }
+        }
+
         if self.capture_output {
             trace!("Capturing the '{}' output", WIX_COMPILER);
             compiler.stdout(Stdio::null());
             compiler.stderr(Stdio::null());
         }
-        compiler
-            .arg("-arch")
-            .arg(&wix_arch.to_string())
-            .arg("-ext")
-            .arg("WixUtilExtension");
+        compiler.arg("-arch").arg(&wix_arch.to_string());
+
+        // Modern wix does not requires `-ext` flags
+        if !self.wix_toolset.is_modern() {
```

Comment:
Here, replace with `if self.wix_toolset.is_legacy()` method. I originally missed the negate, `!`, at the beginning, so I was confused.